program/src/state.rs

```diff
@@ -91,6 +242,21 @@ pub struct HolderRewards {
 #[derive(Clone, Copy, Debug, Default, PartialEq, Pod, ShankAccount, Zeroable)]
 #[repr(C)]
 pub struct HolderRewardsPool {
+    /// The current rewards per token exchange rate.
+    ///
+    /// Stored as a `u128`, which includes a scaling factor of `1e9` to
+    /// represent the exchange rate with 9 decimal places of precision.
+    pub rewards_per_token: u128,
```

Comment:
nit: same here, maybe `accumulated_rewards_per_token` might be clearer?

Reply:
I'd actually push back on this.

Yes, the rate itself is accumulating, but to a holder who hasn't harvested, this is simply the rate at which they can harvest rewards on their token balance, period. They don't know or care it's been accumulating.

To a holder who has harvested, it's the current rate, and they're entitled to the marginal rate (current - last) of the reward pool. They know it's accumulating, but the salient bit is the marginal rate.

I'm inclined to leave it as-is, or possibly rename the two fields to `current_rewards_rate` and `last_harvested_rewards_rate`. What do you think?

Reply:
> to a holder who hasn't harvested, this is simply the rate at which they can harvest rewards on their token balance

Do you mean a holder who hasn't *ever* harvested in the entire lifetime of the rewards pool and also had their tokens minted at the beginning? That's going to be a tiny proportion of use cases, so we shouldn't choose the name based on describing a small subset of usage.

I'm approaching the naming from this situation: if I'm a holder who just received 100 tokens that just harvested rewards, and I see `rewards_per_token` = 10, I might assume that I can get 1000 tokens in rewards, but that's not the case when I call `harvest` again, since I'll get nothing. With `accumulated_rewards_per_token`, I might try to understand a bit more, or I'll be confused and not assume anything, which are both better outcomes.

`rewards_rate` is even less clear, since a rate is with respect to *something*, but what is it with respect to? Time? Tokens? Amount distributed?

Reply:
I guess we'll stick with `rewards_per_token` then :sweat_smile:

Reply:
I can change it to use "accumulated" - I was just trying to come up with an alternative.